kentikapi/internal/resources/base.go

```diff
@@ -4,20 +4,24 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"log"
 
 	"github.com/kentik/community_sdk_golang/kentikapi/internal/api_connection"
 	"github.com/kentik/community_sdk_golang/kentikapi/internal/validation"
 )
 
 // BaseAPI provides marshall/unmarshall + validation functionality for all resource APIs.
 type BaseAPI struct {
-	Transport api_connection.Transport
+	Transport   api_connection.Transport
+	LogPayloads bool
 }
 
 // GetAndValidate retrieves json at "url", unmarshalls and validates against required fields defined in struct tags of "output"
 // output must be pointer to object or nil.
 func (b BaseAPI) GetAndValidate(ctx context.Context, url string, output interface{}) error {
+	b.logPayload("Kentik API request - GET", url, "")
 	responseBody, err := b.Transport.Get(ctx, url)
+	b.logPayload("Kentik API response - GET", url, responseBody)
```

Comment:
Is the response status code logged somewhere?Also, we will have to find a way to avoid logging large responses (e.g. responses to query). In those cases, we should log just the size of the response body.

Reply:
Addressed in https://github.com/kentik/community_sdk_golang/pull/94